src/main/java/it/uniroma3/mat/extendedset/intset/ImmutableConciseSet.java

```diff
@@ -661,6 +692,28 @@ public int compare(WordHolder h1, WordHolder h2)
     return new ImmutableConciseSet(IntBuffer.wrap(retVal.toArray()));
   }
 
+  private static void sort(final WordIterator[] a, final int to, final Comparator<WordIterator> comp)
+  {
+    // Bubble sort, copied from http://stackoverflow.com/a/16089042/648955
+    // Bubble sort seems to be the most efficient sorting algorithm when the number of elements to sort is very
+    // small (in this case mostly 2-4).
+    for (int i = 0; i < to; i++) {
```

Comment:
Yes, but my concern is loading a[j - 1] and a[j] from the array twice. This is very small and hot piece of code so I want it to be as optimal as possible. Probably JIT can perform CSE itself, but I don't rely on it in such questions.